frontend/src/index.tsx

```diff
@@ -30,14 +30,19 @@ const fuse_options = {
 type DownloadResponseOkay = Extract<WebxdcStatusUpdatePayload, { type: 'DownloadOkay' }>
 type UpdateResponse = Extract<WebxdcStatusUpdatePayload, { type: 'Update' }>
 
-function AppInfoModal(item: AppInfoWithState, onDownload: () => void, onForward: () => void, onRemove: () => void) {
+function AppInfoModal(item: AppInfoWithState, onDownload: () => void, onForward: () => void, onRemove: () => void, onDragStart?: (ev: DragEvent) => void) {
   const [isExpanded, setIsExpanded] = createSignal(false)
   const summary = item.description.split('\n')[0]
   const description = item.description.slice(summary.length + 1)
   return (
     <li class="w-full p-3">
       <div class="flex cursor-pointer items-center justify-between gap-2" onClick={() => setIsExpanded(!isExpanded())}>
-        <img src={`data:image/png;base64,${item.image!}`} alt={item.name} class="h-16 w-16 rounded-xl object-cover" />
+        <img 
+          src={`data:image/png;base64,${item.image!}`}
+          alt={item.name}
+          class="h-16 w-16 rounded-xl object-cover"
+          ondragstart={onDragStart}
+          draggable={onDragStart && item.state === AppState.Received} />
```

Comment:
When updating you can also share with the button, that should be possible here aswell.